### PR TITLE
[Generation] fix docs for decoder_input_ids

### LINKED ISSUE #5212: BartConfig wrong decoder_start_token_id?
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): Bart

Language I am using the model on (English, Chinese ...): English

## To reproduce

```
from transformers import BartConfig, BartTokenizer
config = BartConfig.from_pretrained('facebook/bart-large')
tokenizer = BartTokenizer.from_pretrained('facebook/bart-large')
config.decoder_start_token_id
>>> 2
tokenizer.bos_token_id
>>> 0 # != config.decoder_start_token_id
tokenizer.eos_token_id
>>> 2
```

It is misleading in the documentation of the function ```generate````

*decoder_start_token_id=None – (optional) int If an encoder-decoder model starts decoding with a different token than BOS. Defaults to None and is changed to BOS later.*

## Expected behavior

I expect that decoder_start_token_id = tokenizer.bos_token_id, but maybe the model is designed to start decoding with EOS token.

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Fix docs for decoder input ids.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Pinging @sshleifer and @LysandreJik for notification.

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -962,8 +962,9 @@ def generate(
                 `What are attention masks? <../glossary.html#attention-mask>`__
 
             decoder_start_token_id=None: (`optional`) int
-                If an encoder-decoder model starts decoding with a different token than BOS.
-                Defaults to `None` and is changed to `BOS` later.
+                Start token for the decoder. Defaults to `decoder_start_token_id` as defined the model's config or to the `bos_token_id`
```
Comment (patrickvonplaten) [inline src/transformers/modeling_utils.py:965]:
```suggestion
                Start token id for the decoder. Defaults to `decoder_start_token_id` as defined the model's config or to the `bos_token_id`
```

File: src/transformers/modeling_tf_utils.py [hunk 2]
```diff
@@ -642,8 +642,9 @@ def generate(
                 `What are attention masks? <../glossary.html#attention-mask>`__
 
             decoder_start_token_id=None: (`optional`) int
-                If an encoder-decoder model starts decoding with a different token than BOS.
-                Defaults to `None` and is changed to `BOS` later.
+                Start token for the decoder. Defaults to `decoder_start_token_id` as defined the model's config or to the `bos_token_id`
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_utils.py:645]:
```suggestion
                Start token id for the decoder. Defaults to `decoder_start_token_id` as defined the model's config or to the `bos_token_id`
```

File: src/transformers/modeling_tf_utils.py [hunk 3]
```diff
@@ -642,8 +642,9 @@ def generate(
                 `What are attention masks? <../glossary.html#attention-mask>`__
 
             decoder_start_token_id=None: (`optional`) int
-                If an encoder-decoder model starts decoding with a different token than BOS.
-                Defaults to `None` and is changed to `BOS` later.
+                Start token id for the decoder. Defaults to `decoder_start_token_id` as defined the model's config or to the `bos_token_id`
```
Comment (sgugger) [inline src/transformers/modeling_tf_utils.py:645]: The content between single backquotes will render in italics in the doc. It's ok for "optional" but  in most places, argument names are rendered as objects, so I think you should use ` ``decoder_start_token_id`` ` or ` :obj:`decoder_start_token_id` ` and same for bos_token_id

**Review by sshleifer — APPROVED** [review]

File: src/transformers/modeling_tf_utils.py (hunk 3 above)
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_utils.py:645]:
```suggestion
                Start token id for the decoder. Defaults to ``decoder_start_token_id`` as defined the model's config or to the ``bos_token_id``
```

File: src/transformers/modeling_tf_utils.py [hunk 4]
```diff
@@ -642,8 +642,9 @@ def generate(
...
+                if no `decoder_start_token_id` is found in the config.
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_utils.py:646]:
```suggestion
                if no ``decoder_start_token_id`` is found in the config.
```

File: src/transformers/modeling_utils.py [hunk 5]
```diff
@@ -962,8 +962,9 @@ def generate(
                 `What are attention masks? <../glossary.html#attention-mask>`__
 
             decoder_start_token_id=None: (`optional`) int
-                If an encoder-decoder model starts decoding with a different token than BOS.
-                Defaults to `None` and is changed to `BOS` later.
+                Start token id for the decoder. Defaults to `decoder_start_token_id` as defined the model's config or to the `bos_token_id`
```
Comment (patrickvonplaten) [inline src/transformers/modeling_utils.py:965]:
```suggestion
                Start token id for the decoder. Defaults to ``decoder_start_token_id`` as defined the model's config or to the ``bos_token_id``
```

File: src/transformers/modeling_utils.py [hunk 6]
```diff
@@ -962,8 +962,9 @@ def generate(
                 `What are attention masks? <../glossary.html#attention-mask>`__
 
             decoder_start_token_id=None: (`optional`) int
-                If an encoder-decoder model starts decoding with a different token than BOS.
-                Defaults to `None` and is changed to `BOS` later.
+                Start token id for the decoder. Defaults to ``decoder_start_token_id`` as defined the model's config or to the ``bos_token_id``
+                if no `decoder_start_token_id` is found in the config.
```
Comment (patrickvonplaten) [inline src/transformers/modeling_utils.py:966]:
```suggestion
                if no ``decoder_start_token_id`` is found in the config.
```

**Pull request merged**